### PR TITLE
support msgpack canceling

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this pulls the msgpack canceling out of the action bar hiding (which got pulled into marco's pr)

### CODE REVIEW

File: shared/engine/index.js [hunk 1]
```diff
@@ -173,13 +176,27 @@ class Engine {
...
     const {sessionID} = param
+    const {seqid} = response || {seqid: 0}
 
     let once = false
+
+    const cleanup = () => {
+      delete this.sessionIDToResponse[sessionID]
+      delete this.seqIDToWrappedResponse[seqid]
+      if (waitingHandler) {
+        waitingHandler(true, method, sessionID)
```
Comment (MarcoPolo) [inline shared/engine/index.js]: Why do we set this to true?

Comment (chrisnojima) [inline shared/engine/index.js]: ? this is how it works. cleanup is just me moving that code up into a function. both places call true

**Comment by MarcoPolo** [comment]
:shipit:

**Comment by chrisnojima** [comment]
@keybase/react-hackers feel free to merge this, shouldn't do anything bad but it does touch the lower level networking code. or wait till monday :)

**Pull request merged**